### PR TITLE
<Fix> draw; <Update> BillState, Entity

### PR DESCRIPTION
**Pull request by DickyDicky7**

### CODE REVIEW

File: Bill.cpp [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 	self = this;
-	this->vx = 2;
-	this->vy = 7;
-	this->position.x = 50;
-	this->position.y = 150;
-	this->state = new BillNormalState(RIGHT);
+
+	this->vx = 1;
+	this->vy = 1;
+	this->ax = 0.1;
```
Comment (AkiraNoob) [inline Bill.cpp]: what ax means?

File: Bill.cpp [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+	this->ay = 0.1;
```
Comment (AkiraNoob) [inline Bill.cpp]: what ay means

File: Bill.h [hunk 3]
```diff
@@ -54,7 +64,7 @@ class BillState : public State<BillState, Bill>
 
 protected:
 
-	DIRECTION direction;
+	FLOAT time;
```
Comment (AkiraNoob) [inline Bill.h:67]: what is time use for

File: BillRunState.cpp [hunk 4]
```diff
@@ -11,25 +10,58 @@ BillRunState::~BillRunState()
...
+		bill.SetVX(+abs(bill.GetVX()));
+		bill.SetAX(+abs(bill.GetAX()));
+	}
+
+	bill.SetX
+	(
+		bill.GetX() + bill.GetVX() * time + bill.GetAX() * pow(time, 2) / 2
+	);
+
+	if (abs(bill.GetVX()) < +3.00f)
```
Comment (AkiraNoob) [inline BillRunState.cpp:55]: why 3.00f

File: BillRunState.cpp [hunk 5]
```diff
@@ -11,25 +10,58 @@ BillRunState::~BillRunState()
...
+	{
+		bill.SetVX
+		(
+			bill.GetVX() + bill.GetAX() * time
+		);
+	}
+
+	if (time < +2.00f)
```
Comment (AkiraNoob) [inline BillRunState.cpp:65]: why 2.00f

File: Bill.cpp (hunk 1 above)
Comment (DickyDicky7) [reply Bill.cpp]: Ox acceleration

File: Bill.cpp (hunk 2 above)
Comment (DickyDicky7) [reply Bill.cpp]: Oy acceleration

File: Bill.h (hunk 3 above)
Comment (DickyDicky7) [reply Bill.h:67]: Kinematic equations

File: BillRunState.cpp (hunk 4 above)
Comment (DickyDicky7) [reply BillRunState.cpp:55]: Limit the Ox velocity => uniform motion when Ox velocity >= 3. Entity will move too fast if not.

File: BillRunState.cpp (hunk 5 above)
Comment (DickyDicky7) [reply BillRunState.cpp:65]: Entity will move too fast if time >= 2

File: BillRunState.cpp (hunk 4 above)
Comment (AkiraNoob) [reply BillRunState.cpp:55]: comment to code for other member understand

File: BillRunState.cpp (hunk 5 above)
Comment (AkiraNoob) [reply BillRunState.cpp:65]: comment to code for other member understand

Comment (DickyDicky7) [reply BillRunState.cpp:65]: I put the comment in code, please have a review.

File: BillRunState.cpp (hunk 4 above)
Comment (DickyDicky7) [reply BillRunState.cpp:55]: I put the comment in code, please have a review.

**Pull request merged**